packages/redux/action-creator-dispatcher/mixin.runtime-common.js

```diff
@@ -14,14 +14,22 @@ const Dispatcher = withRouter(
       alwaysDispatchActionsOnClient: PropTypes.bool,
       prefetchedOnServer: PropTypes.bool.isRequired,
     };
+    previousPathname = null;
 
     dispatchAll() {
-      if (this.props.location) {
+      const pathnameChanged =
+        this.props.location &&
+        this.props.location.pathname !== this.previousPathname;
+
+      if (pathnameChanged) {
+        this.previousPathname = this.props.location.pathname;
```

Comment:
Seeing this `this.props.location` so many times and also the need for truthy check lets me wonder if we shouldn't make it a little bit simpler by storing the value in a variable:

```suggestion
      const { pathname: currentPathname } = this.props.location || {};

      if (currentPathname !== this.previousPathname) {
        this.previousPathname = pathnameChanged;
```